pkg/splunk/controller/util.go

```diff
@@ -140,6 +140,17 @@ func MergePodSpecUpdates(current *corev1.PodSpec, revised *corev1.PodSpec, name
 			"revised", len(revised.InitContainers))
 		current.InitContainers = revised.InitContainers
 		result = true
+	} else {
+		for idx := range current.InitContainers {
+			// check Image
```

Comment:
Question about the for loop (here as well as later in this function). Once the result has been set to true, is there any reason for continuing parsing through the loop for other containers (or even other if checks in the same iteration)? May be we just break instead?

Reply:
One benefit of continuing to parse the loop is that we log the differences which might help while debugging.

Reply:
Good point. We should really have a dump functionality under debug (at the end of the function, if result is true), if we want to log the differences for every container.

Reply:
Idea is to update the current object, before updating it. So, we update all the specs, wherever there is a change.

Reply:
Ah yes, I missed the updates. 
> Directly writing the revised spec as an update can cause the exception as the resource version and other metadata differ.
Why is this not a concern when `len(current.Containers) != len(revised.Containers)`

Reply:
Had an offline discussion with @sgontla .

Current containers can contain updates which the Operator may not be interested in and can cause unnecessary resets & undesirable behavior. To avoid this, we need to only compare the pre-defined specs of interest and hence the logic. Further we agreed that the condition under which the if statement `len(current.Containers) != len(revised.Containers)` could get executed & its effects need to be evaluated further (for example when Istio is deployed in the same namespace)